diff --git a/node_modules/@tutorialkit/astro/dist/default/components/TutorialContent.astro b/node_modules/@tutorialkit/astro/dist/default/components/TutorialContent.astro
index 8c77b9f..10deb01 100644
--- a/node_modules/@tutorialkit/astro/dist/default/components/TutorialContent.astro
+++ b/node_modules/@tutorialkit/astro/dist/default/components/TutorialContent.astro
@@ -39,10 +39,15 @@ const { Markdown, editPageLink, prev, next } = lesson;
     </div>
   </div>
 
-  <a
-    class="inline-block mt-auto font-size-3.5 underline text-tk-elements-link-secondaryColor hover:text-tk-elements-link-secondaryColorHover"
-    href="https://webcontainers.io/"
-  >
-    {lesson.data.i18n!.webcontainerLinkText}
-  </a>
+  <div class="inline-block mt-auto font-size-3.5">
+    Proudly built by <a
+      class="underline text-tk-elements-link-secondaryColor hover:text-tk-elements-link-secondaryColorHover"
+      href="https://evilmartians.com/"
+      target="_blank"
+    >Evil Martians</a> based on the <a
+      class="underline text-tk-elements-link-secondaryColor hover:text-tk-elements-link-secondaryColorHover"
+      href="https://guides.rubyonrails.org/"
+      target="_blank"
+    >Rails Guides</a>.
+  </div>
 </div>
